File: Practica02/Ejercicios/Ejercicio3/app/src/main/java/com/hector/ejercicio3/MainActivity.java
package com.hector.ejercicio3;

import android.os.Bundle;
import android.view.View;
import android.widget.*;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText editText;
    RadioButton radioDolares, radioSoles;
    Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editText = findViewById(R.id.editText);
        radioDolares = findViewById(R.id.radio0);
        radioSoles = findViewById(R.id.radio1);
        btn = findViewById(R.id.btnConvertir);

        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                convertir();
            }
        });
    }

    private void convertir() {
        String texto = editText.getText().toString();

        if (texto.isEmpty()) {
            Toast.makeText(this, "Ingrese un valor", Toast.LENGTH_SHORT).show();
            return;
        }

        double cantidad = Double.parseDouble(texto);
        double tipoCambio = 3.65;

        if (radioDolares.isChecked()) {
            double soles = cantidad * tipoCambio;
            Toast.makeText(this, cantidad + " dólares = " + soles + " soles", Toast.LENGTH_LONG).show();
        } else {
            double dolares = cantidad / tipoCambio;
            Toast.makeText(this, cantidad + " soles = " + dolares + " dólares", Toast.LENGTH_LONG).show();
        }
    }
}

File: Practica04/Actividades/Actividad3/app/src/main/java/com/hector/actividad3/MainActivity.kt
package com.hector.actividad3

import android.os.Bundle
import android.view.View
import android.widget.TimePicker
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    var timePicker: TimePicker? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        timePicker = findViewById<View?>(R.id.timePicker) as TimePicker
        timePicker!!.setIs24HourView(true)
    }

    fun onClick(view: View?) {
        Toast.makeText(
            getBaseContext(), "Hora seleccionada" +
                    timePicker!!.getCurrentHour() + ":" + timePicker!!.getCurrentMinute(),
            Toast.LENGTH_SHORT
        ).show()
    }
}
